src/transformStateWithClones.js

```diff
@@ -6,8 +6,47 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+
+  const createNewObj = (obj, arr) => {
```

Comment:
Are you really sure that you need this function?

I think we can use spred on line 24 and 30
```js
// 24
results.push(Object.assign(...state, action.extraData));

// 30
results.push( { ...results[results.length - 1] } );
```

What does this function do? Creates a new object? Why can't we create it without it? 
I mean creating a new object it's one line of code and this function is also one line of code but inside we have unnecessary logic. 



if you want to leave this solution, it will be better move `createNewObj`  outside `transformStateWithClones`